rust/lance/src/io/commit.rs

```diff
@@ -200,6 +201,73 @@ async fn migrate_manifest(
     Ok(())
 }
 
+/// Fix schema in case of duplicate field ids.
+///
+/// See test dataset v0.10.5/corrupt_schema
+fn fix_schema(manifest: &mut Manifest) -> Result<()> {
+    let mut field_id_seed = manifest.max_field_id().unwrap_or(-1) + 1;
```

Comment:
Is there any way we can detect things are valid and short-circuit all this computation (might be expensive if there are many columns)?

Reply:
As written, if the dataset is fine, all it will do is collect all the field ids into a `HashSet`. The other data structures are only written to if we detect a duplicate field ids.

We could quickly short circuit if there is only 1 data file per fragment. That will avoid this computation for 90% of datasets out there, since most users aren't adding new columns.